### ISSUE TITLE
Failure to resolve indirect tail calls to vtable slots in some cases

### ISSUE THREAD

**Issue by alexrp** [Vector35/binaryninja-api#3170]
**Version and Platform (required):**
- Binary Ninja Version: 3.1.3482-dev
- OS: Windows
- OS Version: 10.0.25131.1000
- CPU Architecture: x64

**Bug Description:**
Binary Ninja sometimes fails to resolve indirect tail calls to vtable slots.

**Steps To Reproduce:**
No particular steps needed; the issue occurs on a fresh analysis.

**Expected Behavior:**
`this->__base->Malloc(this, size, align)`

**Screenshots:**
![image](https://user-images.githubusercontent.com/44076/172137551-528317e2-5ac6-4a55-9e56-eca073c2d35d.png)
![image](https://user-images.githubusercontent.com/44076/172136746-cfdd346e-7091-48d1-abbe-2be2eeac3c09.png)

**Additional Information:**
I can provide the binary if needed; it's ~52 MB.

```c
struct __packed FMalloc__vtable
{
    void (* __slot1)(struct FMalloc* `this`);
    uint64_t (* QuantizeSize)(struct FMalloc* `this`, uint64_t size);
    void* (* Malloc)(struct FMalloc* `this`, uint64_t size, uint32_t align);
    void* (* Realloc)(struct FMalloc* `this`, void* ptr, uint64_t size, uint32_t align);
    void (* Free)(struct FMalloc* `this`, void* ptr);
    void (* __slot6)(struct FMalloc* `this`);
    void (* __slot7)(struct FMalloc* `this`);
    void (* __slot8)(struct FMalloc* `this`);
    void (* __slot9)(struct FMalloc* `this`);
    void (* __slot10)(struct FMalloc* `this`);
    void (* __slot11)(struct FMalloc* `this`);
    void (* __slot12)(struct FMalloc* `this`);
    void (* __slot13)(struct FMalloc* `this`);
    void (* __slot14)(struct FMalloc* `this`);
    void (* __slot15)(struct FMalloc* `this`);
    void (* __slot16)(struct FMalloc* `this`);
    void (* __slot17)(struct FMalloc* `this`);
    void (* __slot18)(struct FMalloc* `this`);
    void (* __slot19)(struct FMalloc* `this`);
    void (* __slot20)(struct FMalloc* `this`);
    void (* __slot21)(struct FMalloc* `this`);
};

struct __packed FMalloc
{
    struct FMalloc__vtable* __vtable;
};

struct __packed FMallocTBB__vtable
{
    struct FMalloc__vtable __base;
};

struct __packed FMallocTBB
{
    struct FMalloc __base;
    int64_t field_8;
};
```

**Comment by bpotchik** [comment]
@alexrp can you share the binary? You can dm me on slack or email/link to brian@vector35.com

**Comment by alexrp** [comment]
Sure, it's available here: https://drive.google.com/file/d/1LNMS0CdgLw3f2lKHoG8hnZnU5PtAkMdg

**Comment by bpotchik** [comment]
Fixed in `3.1.3575-dev`.
